### PR TITLE
PRP token request behavior

### PR DESCRIPTION
**Pull request by adi130987**

### CODE REVIEW

**Comment by adi130987** [comment]
connects unicef/etools-issues#884

File: pmp/app-config/app-endpoints-behavior.html [hunk 1]
```diff
@@ -1,206 +1,184 @@
...
+    }
+  };
+
+  /** @polymerBehavior EtoolsPmpApp.Behaviors.EndpointsImpl */
+  EtoolsPmpApp.Behaviors.EndpointsImpl = {
+
+    getEndpoint: function(endpointName, data) {
+      let endpoint = JSON.parse(JSON.stringify(EtoolsPmpApp.Endpoints[endpointName]));
+      let needsAuthorizationTokenHeader = ('token' in endpoint);
+      let baseSite = needsAuthorizationTokenHeader
```
Comment (acory) [inline pmp/app-config/app-endpoints-behavior.html]: let needsAuthorizationTokenHeader 
There already is a method for this check that can be reused here : 
authorizationTokenMustBeAdded: function(endpoint) {
      return endpoint && ('token' in endpoint);
    },

Comment (adi130987) [reply pmp/app-config/app-endpoints-behavior.html]: done :)

File: pmp/app-config/app-endpoints-behavior.html [hunk 2]
```diff
@@ -1,206 +1,184 @@
...
-    reportIndicatorsDetails: {
-      url: '/api/indicator/indicator-reports/'
+
+    getTokenEndpointName: function(tokenKey) {
+      return EtoolsPmpApp.EndpointConfig.getTokenEndpoints[tokenKey];
     },
-    reportRequirementsList: {
-      template: '/api/v2/interventions/<%=intervId%>/reporting-periods/'
+
+    prepareRequestOptions: function(endpointName, data) {
```
Comment (acory) [inline pmp/app-config/app-endpoints-behavior.html]: more suggestive name addTokenToRequestOptions

Comment (adi130987) [reply pmp/app-config/app-endpoints-behavior.html]: done :)

**Pull request merged**